### PR TITLE
[QT] Add password mismatch warnings

### LINKED ISSUE #1605: [Feature request] Prevent going next step if initial wallet encryption passphrases do not match
Although it would return an error if passphrases do not match while saving it, wallet allows proceeding first step with non-matching passphrases. Feature request to prevent it in the first step.

encryptionBug1

encryptionBug2

### PR DESCRIPTION
**Pull request by JSKitty**
This closes #1605 by adding a warning message & deactivating the "OK" button when the two passphrase fields do not match eachother.

image

As this warning is in the same area as the caps lock warning, the two have been merged into a single "passWarningLabel" which "stacks" warnings ontop of eachother, so both warnings can be shown independently, or both at once.

image

This PR ensures a user cannot proceed via the "OK" button until:
1. The new passphrase fields are not empty.
2. The new passphrase fields share identical texts.
3. (If changing password only) The **old** passphrase field is not empty.

### CODE REVIEW

File: src/qt/askpassphrasedialog.cpp [hunk 1]
```diff
@@ -272,9 +270,24 @@ bool AskPassphraseDialog::event(QEvent* event)
...
+        if (fCapsLock && !validPassphrases) warningStr += "<br>";
+        if (!validPassphrases) warningStr += tr("WARNING: Passphrases do not match!");
+
+        if (warningStr.isEmpty()) {
+            ui->passWarningLabel->clear();
+            ui->passWarningLabel->setVisible(false);
+        } else {
+            ui->passWarningLabel->setText(warningStr);
+            ui->passWarningLabel->setVisible(true);
+        }
```
Comment (furszy) [inline src/qt/askpassphrasedialog.cpp:289]: same code here and in the eventFilter method, better to create a separate method.

File: src/qt/askpassphrasedialog.cpp [hunk 2]
```diff
@@ -272,9 +270,24 @@ bool AskPassphraseDialog::event(QEvent* event)
...
-            fCapsLock ? ui->capsLabel->setText(tr("WARNING: The Caps Lock key is on!")) : ui->capsLabel->clear();
         }
+
+        // Merge warning labels together if there's two warnings
+        bool validPassphrases = false;
+        validPassphrases = ui->passEdit2->text() == ui->passEdit3->text();
+        QString warningStr = "";
+        if (fCapsLock) warningStr += tr("WARNING: The Caps Lock key is on!");
+        if (fCapsLock && !validPassphrases) warningStr += "<br>";
+        if (!validPassphrases) warningStr += tr("WARNING: Passphrases do not match!");
```
Comment (furszy) [inline src/qt/askpassphrasedialog.cpp:281]: To not duplicate the "Warning" word twice, thinking that we could try the following organization:
```
Warning:
* The caps lock key is on!
* Passphrases do not match!
```

File: src/qt/askpassphrasedialog.cpp (hunk 1 above)
Comment (JSKitty) [reply src/qt/askpassphrasedialog.cpp]: Refactored into `getWarnings()`

File: src/qt/askpassphrasedialog.cpp (hunk 2 above)
Comment (JSKitty) [reply src/qt/askpassphrasedialog.cpp]: Good idea, looks much nicer on the eyes this way; done.

**Comment by JSKitty** [comment]
Added suggestions and updated the PR description with new images of the compiled PR.

**Comment by random-zebra** [comment]
Good job :+1: works as intended and code looks good. Only minor non-blocking nit would be rename `getWarnings()`, since it  doesn't return the warning string, but rather update the label (maybe `updateWarningsLabel()` ?).

**Comment by JSKitty** [comment]
Renamed `getWarnings()` to `updateWarningsLabel()`

**Pull request merged**